### PR TITLE
Switch from using class variables to using the rails cache in the Concept model

### PR DESCRIPTION
**Pull request by bethanyaconnor**

### CODE REVIEW

File: dashboard/app/models/concept.rb [hunk 1]
```diff
@@ -27,15 +27,19 @@ class Concept < ActiveRecord::Base
...
+    Rails.cache.fetch('concepts/all') do
+      Script.all
+    end
   end
 
   def self.expire_cache
-    @@all_cache = nil
-    @@name_cache = nil
+    Rails.cache.delete_if {|key, _| key.start_with?("concepts/")}
   end
```
Comment (Hamms) [inline dashboard/app/models/concept.rb]: it doesn't look like we actually use this method; should we just remove it?

File: dashboard/app/models/concept.rb [hunk 2]
```diff
@@ -27,15 +27,19 @@ class Concept < ActiveRecord::Base
 
   def self.by_name(name)
     (@@name_cache ||= Concept.all.index_by(&:name))[name].try(:id)
+    Rails.cache.fetch('concepts/names/{name}') do
```
Comment (Hamms) [inline dashboard/app/models/concept.rb]:
```suggestion
    Rails.cache.fetch("concepts/names/#{name}") do
```

File: dashboard/app/models/concept.rb (hunk 1 above)
Comment (bethanyaconnor) [reply dashboard/app/models/concept.rb]: There's a commented out line calling this but I'm guessing that it's not important if we haven't fixed it in 5+ years. :)

File: dashboard/app/models/concept.rb (hunk 2 above)
Comment (bethanyaconnor) [reply dashboard/app/models/concept.rb]: Done.

File: dashboard/app/models/concept.rb [hunk 3]
```diff
@@ -22,20 +22,18 @@ class Concept < ActiveRecord::Base
   include Seeded
   has_and_belongs_to_many :levels
   belongs_to :video
-  # Can't call static from filter. Leaving in place for fixing later
-  #after_save :expire_cache
 
   def self.by_name(name)
     (@@name_cache ||= Concept.all.index_by(&:name))[name].try(:id)
```
Comment (Hamms) [inline dashboard/app/models/concept.rb]: this line should probably be removed

File: dashboard/app/models/concept.rb [hunk 4]
```diff
@@ -22,20 +22,18 @@ class Concept < ActiveRecord::Base
...
 
   def self.cached
-    @@all_cache ||= Concept.all
-  end
-
-  def self.expire_cache
-    @@all_cache = nil
-    @@name_cache = nil
+    Rails.cache.fetch("concepts/all") do
+      Script.all
```
Comment (Hamms) [inline dashboard/app/models/concept.rb]: should this be Concept.all?

Comment (bethanyaconnor) [reply dashboard/app/models/concept.rb]: Ugh this is why copy/pasting is bad :(

**Review by Hamms — APPROVED** [review]

**Pull request merged**